drivers/android/thread.rs

```diff
@@ -241,23 +246,13 @@ impl Thread {
     pub(crate) fn new(id: i32, process: Arc<Process>) -> Result<Arc<Self>> {
         let return_work = Arc::try_new(ThreadError::new(InnerThread::set_return_work))?;
         let reply_work = Arc::try_new(ThreadError::new(InnerThread::set_reply_work))?;
-        let mut thread = Pin::from(UniqueArc::try_new(Self {
+        let thread = UniqueArc::pin_init::<core::convert::Infallible>(pin_init!(Self {
```

Comment:
This could also just be an `Arc` as the update performed below is through a lock.